python/across_api/base/footprint.py

```diff
@@ -0,0 +1,43 @@
+# Copyright © 2023 United States Government as represented by the
+# Administrator of the National Aeronautics and Space Administration.
+# All Rights Reserved.
+
+
+from astropy import units as u  # type: ignore[import]
+from astropy.coordinates.representation import (  # type: ignore[import]
+    CartesianRepresentation,
+    UnitSphericalRepresentation,
+)
+from astropy.coordinates.matrix_utilities import rotation_matrix  # type: ignore[import]
+
+
+class Footprint:
+    polygon: list = []
+
+    def __init__(self, polygon: list) -> None:
+        self.polygon = polygon
+
+    def project(self, ra, dec, pos_angle):
+        if pos_angle is None:
+            pos_angle = 0.0
+
+        projected_footprint = (
+            UnitSphericalRepresentation(
+                u.Quantity([pt[0] for pt in self.polygon], u.deg),
+                u.Quantity([pt[1] for pt in self.polygon], u.deg),
+            )
+            .represent_as(CartesianRepresentation)
+            .transform(rotation_matrix(-1.0 * pos_angle * u.deg, axis="x"))
+            .transform(rotation_matrix(dec * u.deg, axis="y"))
+            .transform(rotation_matrix(-1.0 * ra * u.deg, axis="z"))
+            .represent_as(UnitSphericalRepresentation)
+        )
+
+        projected_ra = [
+            round(lon.to(u.deg).value, 3) for lon in projected_footprint.lon
+        ]
+        projected_dec = [
+            round(lat.to(u.deg).value, 3) for lat in projected_footprint.lat
+        ]
+
+        return list(zip(projected_ra, projected_dec))
```

Comment:
Again, I suggest using sky offset frames. Roughly like this:

```suggestion
    def project(self, center: SkyCoord, pos_angle: u.Quantity[u.deg] = 0 * u.deg):
        representation = (
            UnitSphericalRepresentation(
                u.Quantity([pt[0] for pt in self.polygon], u.deg),
                u.Quantity([pt[1] for pt in self.polygon], u.deg),
            )
        )
        return SkyCoord(representation, frame=center.skyoffset_frame(pos_angle).icrs
```

Reply:
Implemented in recent push. Makes the `footprint.py` file very concise.